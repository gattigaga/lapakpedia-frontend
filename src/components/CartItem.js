import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { toCurrency, cutText } from "helpers/formatter";
import Counter from "./Counter";

const Container = styled.div`
  display: flex;
  padding: 18px;
  border-bottom: 1px solid #ddd;
  position: relative;
`;

const Image = styled.img`
  width: 80px;
  height: 80px;
  object-fit: cover;
  margin-right: 18px;
  box-sizing: border-box;
  border: 1px solid #ddd;
`;

const ContentContainer = styled.div`
  flex: 1;
`;

const Name = styled.h5`
  font-family: Roboto;
  font-size: 16px;
  color: black;
  margin: 0px;
  margin-bottom: 18px;
`;

const Price = styled.p`
  font-size: 24px;
  font-family: Roboto;
  color: #ff8800;
  margin: 0px;
  margin-right: 32px;
`;

export const CloseButton = styled.span`
  font-size: 24px;
  cursor: pointer;
  padding: 8px;
  position: absolute;
  top: 4px;
  right: 4px;
`;

const CartItem = ({
  onClickClose,
  onClickMinus,
  onClickPlus,
  onChange,
  maximum,
  value,
  name,
  image,
  price
}) => {
  const totalWords = name.split(" ").length;

  return (
    <Container>
      <Image src={image} />
      <ContentContainer>
        <Name>{totalWords > 5 ? cutText(5, name) : name}</Name>
        <Counter
          onClickPlus={onClickPlus}
          onClickMinus={onClickMinus}
          onChange={onChange}
          maximum={maximum}
          value={value}
        />
      </ContentContainer>
      <Price>$ {toCurrency(price)}</Price>
      <CloseButton onClick={onClickClose}>&times;</CloseButton>
    </Container>
  );
};

CartItem.propTypes = {
  onClickClose: PropTypes.func,
  onClickMinus: PropTypes.func,
  onClickPlus: PropTypes.func,
  onChange: PropTypes.func,
  maximum: PropTypes.number,
  value: PropTypes.number,
  name: PropTypes.string,
  image: PropTypes.string,
  price: PropTypes.number
};

export default CartItem;
