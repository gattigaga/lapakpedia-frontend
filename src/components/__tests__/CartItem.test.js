import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import CartItem, { CloseButton } from "components/CartItem";

describe("CartItem", () => {
  jest.mock("../Counter", () => "Counter");

  const setup = propOverrides => {
    const props = {
      ...propOverrides,
      value: 10,
      maximum: 15,
      image: require("assets/images/dummy240x240.png"),
      name: "MacBook Pro 2018",
      price: 2800,
      onClickClose: jest.fn(),
      onClickMinus: jest.fn(),
      onClickPlus: jest.fn(),
      onChange: jest.fn()
    };

    const wrapper = shallow(<CartItem {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should call 'onClickClose' callback while close button clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(CloseButton).simulate("click");

    expect(props.onClickClose).toBeCalled();
  });
});
