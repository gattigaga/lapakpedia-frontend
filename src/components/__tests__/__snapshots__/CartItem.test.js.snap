// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should renders default 1`] = `
<styled.div>
  <styled.img
    src="dummy240x240.png"
  />
  <styled.div>
    <styled.h5>
      MacBook Pro 2018
    </styled.h5>
    <Counter
      maximum={15}
      onChange={[Function]}
      onClickMinus={[Function]}
      onClickPlus={[Function]}
      value={10}
    />
  </styled.div>
  <styled.p>
    $ 
    2,800
  </styled.p>
  <styled.span
    onClick={[Function]}
  >
    Ã—
  </styled.span>
</styled.div>
`;
